src/main/java/seedu/guest/model/guest/Room.java

```diff
@@ -0,0 +1,56 @@
+package seedu.guest.model.guest;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.guest.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents the Guest's room
+ * Guarantees: immutable; is valid as declared in {@link #isValidRoom(String)}
+ */
+public class Room {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Room should only contain alphanumeric characters and a hyphen. "
+            + "Should it contain a hyphen, it must be in between 2 alphanumeric characters. "
+            + "There should also be no spacings in between. "
+            + "E.g 05-73 or Aloha-5 are accepted.";
+
+    // Room contains alphanumeric characters with an optional hyphen only in between, and don't start with a hyphen
+    private static final String VALIDATION_REGEX = "^[A-Za-z0-9]+(?:-?[A-Za-z0-9]+)+$";
```

Comment:
Can I check if this VALIDATION_REGEX accepts "09-09-09"? If yes, the MESSAGE_CONSTRAINTS description might have missed that part out?

Reply:
Indeed, added!